twitch4j/src/main/java/com/github/twitch4j/TwitchClientHelper.java

```diff
@@ -308,50 +331,49 @@ public TwitchClientHelper(TwitchHelix twitchHelix, EventManager eventManager, Sc
                 return false;
             }
         };
-    }
+        this.clipEventTask = channelId -> {
+            // check clip creations
+            boolean nextRequestCanBeImmediate = false;
+
+            final ChannelCache currentChannelCache = channelInformation.get(channelId, c -> new ChannelCache());
+            final AtomicReference<Instant> windowStart = currentChannelCache.getClipWindowStart();
+            final Instant startedAt = windowStart.get();
+            final Instant now = Instant.now();
+
+            if (startedAt == null) {
+                // initialize clip query window started_at
+                nextRequestCanBeImmediate = windowStart.compareAndSet(null, now);
```

Comment:
This is fine, restarts aren't very frequent and i'd prefer avoiding duplicates.

It's simple to store the last timestamp and pull "missing" clips from the downtime once on startup if needed. If we want to make this easier we could also allow to overwrite/set the "last timestamp" on the initial `enableClipEventListener` call - though this would be subject to the 20 * 100 limit (but should be fine for a restart).